### ISSUE TITLE
Option to include linked time zones

### ISSUE THREAD

**Issue by caseyjhol** [gilmoreorless/moment-timezone-data-webpack-plugin#54]
The "America/New_York" time zone has the following alternative names:
```
			"America/New_York|America/Detroit",
			"America/New_York|America/Fort_Wayne",
			"America/New_York|America/Indiana/Indianapolis",
			"America/New_York|America/Indiana/Marengo",
			"America/New_York|America/Indiana/Petersburg",
			"America/New_York|America/Indiana/Vevay",
			"America/New_York|America/Indiana/Vincennes",
			"America/New_York|America/Indiana/Winamac",
			"America/New_York|America/Indianapolis",
			"America/New_York|America/Iqaluit",
			"America/New_York|America/Kentucky/Louisville",
			"America/New_York|America/Kentucky/Monticello",
			"America/New_York|America/Louisville",
			"America/New_York|America/Montreal",
			"America/New_York|America/Nassau",
			"America/New_York|America/Nipigon",
			"America/New_York|America/Pangnirtung",
			"America/New_York|America/Port-au-Prince",
			"America/New_York|America/Thunder_Bay",
			"America/New_York|America/Toronto",
			"America/New_York|Canada/Eastern",
			"America/New_York|EST5EDT",
			"America/New_York|US/East-Indiana",
			"America/New_York|US/Eastern",
			"America/New_York|US/Michigan",
```

However, when using:
```
new MomentTimezoneDataPlugin({
    matchCountries: ['US'],
}),
```
or
```
new MomentTimezoneDataPlugin({
    matchZones: ['America/New_York'],
}),
```
None of the alternative names show up in `moment.tz.names()`, and attempting to use any of those zones results in an error. I see there is some existing code that appears to handle links, but it apparently doesn't do what I was expecting.

It would be great if there was an option to include linked time zones as well.

**Comment by gilmoreorless** [comment]
I agree this is a slightly confusing scenario. (I even had to go and double-check my own code just to make sure I properly understood what was happening.)

First, here's why the plugin (currently) doesn't include `US/Eastern`:

1. The `matchZones` option will include only zones that match the provided value(s). If a zone is defined in `moment-timezone` as a link, then the zone it points to is also included. So asking for `US/Eastern` will also include `America/New_York`, but not the other way around. This was done because some zones have many, many links (e.g. `America/Puerto_Rico` has 20 links pointing to it).
2. The `matchCountries` option chooses which zones are included based on the `countries` data provided by `moment-timezone`. For `US`, [this is 29 zones](https://github.com/moment/moment-timezone/blob/develop/data/meta/2022e.json#L1861-L1895).
    1. Those entries originally come from the [`zone.tab`](https://data.iana.org/time-zones/data/zone.tab) and [`zone1970.tab`](https://data.iana.org/time-zones/data/zone1970.tab) files in the IANA time zone database. `US/Eastern` doesn't appear in any of those lists, mainly because it's using an old, long-deprecated naming format that the tzdb maintainers would like people to move away from.
    2. As with `matchZones`, there is some link handling here, but only if the links are found in that primary `countries` list.

I'll have a think about adding an option to include all links that point to the defined zones. It would probably be useful for other use cases too. In the meantime I think adding the extra `^US/` matcher like you have is the best workaround.

I've also created #55 to add more documentation about how zones are included.
